media-libs/libyuv/libyuv-9999.ebuild

```diff
@@ -0,0 +1,50 @@
+# Copyright 2025 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit cmake
+
+DESCRIPTION="Open source project that includes YUV scaling and conversion functionality."
+HOMEPAGE="https://chromium.googlesource.com/libyuv/libyuv"
+
+if [[ ${PV} == "9999" ]] ; then
+	EGIT_REPO_URI="https://chromium.googlesource.com/libyuv/libyuv.git"
+	inherit git-r3
+else
+	# to diff against upstream (apparently not stable):
+	# https://chromium.googlesource.com/libyuv/libyuv.git/+archive/${commit}.tar.gz
+	MYTAG="0.0.1904.20250204"
+	SRC_URI="https://salsa.debian.org/debian/libyuv/-/archive/upstream/${MYTAG}/libyuv-upstream-${MYTAG}.tar.bz2 -> ${P}.tar.bz2"
+	S="${WORKDIR}/libyuv-upstream-${MYTAG}"
+	KEYWORDS="~amd64"
+fi
+
+LICENSE="BSD"
+SLOT="0/${PV}"
+# TODO(NRK): enable tests
+RESTRICT="test"
+
+RDEPEND=">=media-libs/libjpeg-turbo-3.0.0"
+BDEPEND="${RDEPEND}"
+
+PATCHES=(
+	"${FILESDIR}/0001-fix-install-dirs.patch"
+	"${FILESDIR}/0002-disable-static-library.patch"
+	"${FILESDIR}/0003-disable-test-tools.patch"
+)
+
+src_configure() {
+	mycmakeargs=(
+		-DUNIT_TEST=OFF
+	)
+	cmake_src_configure
+}
+
+src_install() {
+	cmake_src_install
+
+	insinto /usr/"$(get_libdir)"/pkgconfig
+	sed "s|@LIBDIR@|$(get_libdir)|;s|@VERSION@|${PV}|;" "${FILESDIR}/libyuv.pc" |
```

Comment:
Hmmm, should the live ebuild set the version to 9999 here? That seems wrong. Grep it out of `include/libyuv/version.h` instead?

Reply:
Upstream doesn't have a pkgconfig file at all. Neither does Debian. Why should we add one?

If one did get added it should be configured by cmake rather than via a sed in src_install. That is (again) best done upstream.

Reply:
> Neither does Debian.

Huh, I thought they did. Fedora does though.

> Why should we add one?

If not then patching imlib2 would be needed.

> That is (again) best done upstream.

Agreed. But I'm not interested in making a google account to sign CLA in order to do so.

Reply:
> If not then patching imlib2 would be needed.
> 
> > That is (again) best done upstream.
> 
> Agreed. But I'm not interested in making a google account to sign CLA in order to do so.

You could instead submit an upstream patch to imlib2, which would make imlib2 build with libyuv support on distros such as Debian which don't have that pkg-config file but do have libyuv. No CLA required in that case.

Reply:
> You could instead submit an upstream patch to imlib2, which would make imlib2 build with libyuv support on distros such as Debian which don't have that pkg-config file but do have libyuv. No CLA required in that case.

Not super enthusiastic about patching autoconf stuff :)

I opened an issue letting imlib2 know about it, though: https://git.enlightenment.org/old/legacy-imlib2/issues/31

Reply:
@xen0n I wonder, since I happened to notice you have a Google gerrit accound were recently submitting patches for libyuv, would you be interested in e.g. contributing an upstream pkg-config file, fixing the install directory, etc?